frontend/src/metabase-lib/v1/parameters/utils/targets.ts

```diff
@@ -58,13 +60,33 @@ export function getParameterTargetField(
   }
 
   if (isConcreteFieldReference(fieldRef)) {
-    const fieldId = fieldRef[1];
-    const resultMetadata = question.getResultMetadata();
-    const fieldMetadata = resultMetadata.find(field => field.id === fieldId);
-    return (
-      metadata.field(fieldId, fieldMetadata?.table_id) ??
-      metadata.field(fieldId)
+    const query = question.query();
+    const stageIndex = -1;
+    const columns = Lib.visibleColumns(query, stageIndex);
+
+    // query and metadata are not available
+    if (columns.length === 0) {
+      const [_, fieldId] = fieldRef;
+      return metadata.field(fieldId);
+    }
+
+    const [columnIndex] = Lib.findColumnIndexesFromLegacyRefs(
+      query,
+      stageIndex,
+      columns,
+      [fieldRef],
     );
+    if (columnIndex < 0) {
+      return null;
+    }
+
+    const column = columns[columnIndex];
+    const fieldValuesInfo = Lib.fieldValuesSearchInfo(query, column);
+    if (fieldValuesInfo.fieldId == null) {
```

Comment:
this is also not clear, looks like a pseudocode to me. can we have some comment explaining what's going on?

Reply:
Added a comment